client/Components/Reservation.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Costs from './Costs';
+import Form from './Form';
+import TopRow from './TopRow';
+import styles from './Reservation.css';
+import { fetchData } from '../store';
+
+const mapStateToProps = (state) => {
+  const { checkin, checkout } = state.dateReducer;
+  let duration = 0;
+  if (checkin && checkout) {
+    const checkinDate = checkin.slice(-2);
+    const checkoutDate = checkout.slice(-2);
+
+    duration = Number.parseInt(checkoutDate, 10) - Number.parseInt(checkinDate, 10);
+  }
+  const { room } = state.roomReducer;
+  return { duration, room };
+};
+
+const mapDispatchToProps = {
+  fetchData,
+};
+
+class ConnectedReservation extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const { fetchData } = this.props;
+    fetchData();
+    // try {
+    //   const data = await this.fetchData();
+    //   this.setState(data);
+    // } catch (err) {
+    //   console.log(err);
+    // }
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    e.persist();
+    console.log(e);
```

Comment:
Assuming this is currently a stub

Reply:
It is a stub that may ultimately be removed. I may need to keep it just for Reservation button interactivity, but actually making a reservation is not a feature we're implementing. I just need the button to not cause a refresh.